utils/load_image.py

```diff
@@ -0,0 +1,90 @@
+#%%
+import tarfile
+from io import BytesIO
+from PIL import Image
+import os
+from pathlib import Path
+import pandas as pd
+
+
+basefolder_loc = Path(__file__).parents[1]
+TARFILE = tarfile.open(
+    os.path.join(basefolder_loc, "1.download-data", "data", "training.tar.gz"), "r:gz",
+)
+
+
+def load_img(
+    target: str = "adrenoceptor",
+    plate: str = "P1",
+    cell_id: int = 1,
+    replicate: int = 1,
+    wellName: str = "C10",
+    field: int = 1,
+) -> Image:
+    """
+    Find the single image that matches on all given parameters.
+
+    Returns: image
+
+    WARNING:
+    Use this only if you need to load a single image.
+    Every time you run this function it loops over the whole zip file.
```

Comment:
This is great docstring, but I am wondering if there any reason why it needs to loop over the whole zip file?

Should there be a processing step where all single cell images are extracted first?

Reply:
We can first unzip the files but in that case the user needs more memory and load more from its cold to hot memory. I haven't tested if it is really slower but I think so